### DISCUSSION TITLE
Any workaround around old libraries ? "command: use_2to3 is invalid"

### DISCUSSION THREAD

**Discussion by liran-cohen-hs** [pdm-project/pdm#2159] [Q&A] [answered]
As part of the migration to PDM, we are handling some legacy projects and libraries
one example is [suds-jurko](https://pypi.org/project/suds-jurko/) 
when trying to run `pdm add suds-jurko` it fails with this error:

> pdm.exceptions.BuildError: Build backend raised error: Showing the last 10 lines of the build output:
error in suds-jurko setup command: use_2to3 is invalid.

I tried all the different build backends listed in [this](https://pdm.fming.dev/latest/reference/build/) page

I guess it's a matter of using old setuptools

any workaround of making it work ? (other than the obvious thing we should do and upgrade this library to something newer :))

Thanks !

**Comment by frostming** [comment, marked as answer]
It is because the `use_2to3` functionality is [no longer supported](https://setuptools.pypa.io/en/latest/history.html#v58-0-2) since `setuptools==58`.

You can workaround this with following:

```
pdm add "setuptools<58"
pdm add --no-isolation suds-jurko
```
`--no-isolation` will reuse the libraries already installed in the project environment(in this case `setuptools`), instead of creating an isolated environment and install build requirements from empty.

I can confirm it works with Python 3.11

**Reply by liran-cohen-hs** [reply]
Thanks
seems to be working :)
